test/service/test_pagination.py

```diff
@@ -34,22 +39,33 @@ def assert_page1_correct(self, json_response):
         :param json_response: A JSON response dictionary
         :return:
         """
-        if 'search_before' in json_response['pagination']:
```

Comment:
I'd feel more confident in the fact that your changes don't break anything if the test changes weren't so invasive. For now we should stick with `search_before` and `search_after` being the pivots, not `previous` and `next`. After all you're adding response elements so the test code should only have additions, no deletions, or minimal deletions.